gravitee-node-kubernetes/src/main/java/io/gravitee/node/kubernetes/keystoreloader/KubernetesSecretKeyStoreLoader.java

```diff
@@ -123,21 +122,43 @@ protected Flowable<Secret> watch() {
     @Override
     protected Completable loadKeyStore(Secret secret) {
         final Map<String, String> data = secret.getData();
-        final KeyStore keyStore;
+        KeyStore keyStore = null;
 
         if (secret.getType().equals(KUBERNETES_TLS_SECRET)) {
-            keyStore =
-                KeyStoreUtils.initFromPem(
-                    new String(Base64.getDecoder().decode(data.get(KUBERNETES_TLS_CRT))),
-                    new String(Base64.getDecoder().decode(data.get(KUBERNETES_TLS_KEY))),
-                    options.getKeyStorePassword(),
-                    secret.getMetadata().getName()
-                );
+            keyStore = secretToKeyStore(secret);
         } else if (secret.getType().equals(KUBERNETES_OPAQUE_SECRET)) {
             if (options.getKeyStoreType().equalsIgnoreCase(CERTIFICATE_FORMAT_PEM)) {
                 return Completable.error(
                     new IllegalArgumentException("Pem format is not supported with opaque secret, use kubernetes tls secret instead.")
                 );
+            } else if (options.getKeyStoreType().equalsIgnoreCase(CERTIFICATE_FORMAT_PEM_REGISTRY)) {
```

Comment:
Considering `http.ssl.keystore.kubernetes` is an array of secret references that end up being merged into a single keystore, I don't see the value of this feature. Looks like it comes from GKO project. Why not adding this capability to the helm chart to load a list of secrets rather than creating a Secret (that is not a secret) containing references to secrets. On top of that they cannot reside into another namespace than the 'registry secret' itself.

Reply:
The problem is that we don't know the secret names up-front and they will be added/updated/deleted in the run-time. So nothing is clear up front about the secretes. And yes, the pem-registry will be in the same namespace, since you can't refernce a secret outside the namespace for the ingress TLS

Reply:
Ok, I get it. Why not in a config-map then ?

Reply:
It is not an issue, we can put it in a ConfigMap

Reply:
I changed the pem-registry to a ConfigMap